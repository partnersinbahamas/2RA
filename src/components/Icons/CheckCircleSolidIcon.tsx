import React from 'react';
import PropTypes from 'prop-types';

import { SmallSvg } from './Svg';
import { TIcon } from '.';

const CheckCircleSolidIcon: React.FC<TIcon> = ({
  className,
  color,
  size,
  style,
  withBox,
}) => {
  switch (size) {
    case 'SMALL':
    default:
      return (
        <SmallSvg
          xmlns="http://www.w3.org/2000/svg"
          viewBox={withBox ? '-2 -2 20 20' : '0 0 16 16'}
          style={style}
          fill={color}
          className={className}
        >
          <path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M8 0C12.41 0 16 3.563 16 7.99C16 12.405 12.41 16 8 16C3.59 16 0 12.405 0 7.99C0 3.595 3.59 0 8 0ZM11.493 4.732L6.595 9.629L4.51 7.548L3.066 8.953L6.621 12.429L12.938 6.156L11.494 4.732H11.493Z"
          />
        </SmallSvg>
      );
  }
};

CheckCircleSolidIcon.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string,
  style: PropTypes.any,
  withBox: PropTypes.bool,
};

CheckCircleSolidIcon.defaultProps = {
  className: '',
  color: 'currentcolor',
  size: 'LARGE',
  style: {},
  withBox: false,
};

export default CheckCircleSolidIcon;
