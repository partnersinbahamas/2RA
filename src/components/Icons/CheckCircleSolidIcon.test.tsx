import React from 'react';
import { render } from '@testing-library/react';

import CheckCircleSolidIcon from './CheckCircleSolidIcon';

describe('CheckCircleSolidIcon', () => {
  it('renders small', () => {
    const { container } = render(<CheckCircleSolidIcon size="SMALL" />);
    expect(container).toMatchSnapshot();
  });

  it('renders default', () => {
    const { container } = render(<CheckCircleSolidIcon />);
    expect(container).toMatchSnapshot();
  });
});
