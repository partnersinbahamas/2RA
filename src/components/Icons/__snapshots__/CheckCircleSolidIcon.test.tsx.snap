// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckCircleSolidIcon renders default 1`] = `
<div>
  <svg
    class="sc-guDLey kBBjNb"
    fill="currentcolor"
    viewBox="0 0 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M8 0C12.41 0 16 3.563 16 7.99C16 12.405 12.41 16 8 16C3.59 16 0 12.405 0 7.99C0 3.595 3.59 0 8 0ZM11.493 4.732L6.595 9.629L4.51 7.548L3.066 8.953L6.621 12.429L12.938 6.156L11.494 4.732H11.493Z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`CheckCircleSolidIcon renders small 1`] = `
<div>
  <svg
    class="sc-guDLey kBBjNb"
    fill="currentcolor"
    viewBox="0 0 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M8 0C12.41 0 16 3.563 16 7.99C16 12.405 12.41 16 8 16C3.59 16 0 12.405 0 7.99C0 3.595 3.59 0 8 0ZM11.493 4.732L6.595 9.629L4.51 7.548L3.066 8.953L6.621 12.429L12.938 6.156L11.494 4.732H11.493Z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;
